### PR TITLE
[CI] Add NuGet MSI conversion and VS insert stage

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/yaml-templates/pull/117
Context: https://github.com/xamarin/yaml-templates/pull/121

Updates `BundledVersions.targets` and `WorkloadManifest.json` exclude
package semver metadata, as this is excluded in both Android and maciOS.
The .msi conversion tooling also does not process these semver metadata
correctly.

A new CI job and stage have been added to facilitate the Visual Studio
installer insertion process.

The first job named "Convert NuGet to MSI" will convert signed packages
to .msi installers, and generate Visual Studio manifests for the .msi
installers.  The .msi files are signed and converted back to .nupkg
files for the Windows command line install scenario.  These packages are
uploaded as pipeline artifacts named `vsdrop-signed`, and 
`vs-msi-nugets`.

The new `vs-workload.props` file contains version information and other
metadata used by the convert job to generate a Visual Studio manifest.

The new "VS Insertion" stage starts with a [manual validation task][0].
This task will pause execution and wait for someone to click a "Resume"
or "Reject" button that will appear on the pipeline UI.  This task is
configured to be rejected after waiting for two days, but it can be
manually re-ran any time after being rejected to trigger a VS insertion
for an older build.

If the manual validation task is approved, a VS Drop will be created
containing all .NET 6 .msi files.  This Drop URL can then be used to
update our component versions in Visual Studio.  This last piece is
currently manual as we will initially be introducing new components,
however we should be able to automate VS PR creation in the future.

[0]: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/manual-validation?view=azure-devops&tabs=yaml

### CODE REVIEW

**Comment by rmarinho** [comment]
Do we want to merge to yaml-templates before taking this ?

File: eng/automation/vs-workload.template.props
Comment (jonathanpeppers) [inline eng/automation/vs-workload.template.props:29]: How short do these have to be? Would a name like `Essentials.Runtime` still be too long?

**Comment by jonathanpeppers** [comment]
> Do we want to merge to yaml-templates before taking this ?

Yes, I think we need this first: https://github.com/xamarin/yaml-templates/pull/117

It looks like there are some temporary changes to bump iOS & Android here, too.

File: eng/automation/vs-workload.template.props
Comment (pjcollins) [reply eng/automation/vs-workload.template.props:29]: We have a 64 character limit from what I've seen in my testing, but this includes the full version string that we add to file names. I am currently looking at trimming the version information instead this name string, as the version information will still be associated with the pack even if it is not in the file name.

Comment (pjcollins) [reply eng/automation/vs-workload.template.props]: We now replace the semver version in the pack names with the shorter four part version that is used for the generated VS components. This has a more consistent length and should be all that is needed to avoid path issues during the VS build.

**Comment by pjcollins** [comment]
Latest test run triggered here https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4998323&view=results.

**Comment by pjcollins** [comment]
@mattleibow @rmarinho when you have a chance to review, I think we can land this for now and then take any future incremental changes as needed for preview 7.

**Review by mattleibow — APPROVED** [review]

**Comment by pjcollins** [comment]
Let's hold off on this for now, we have more yaml-templates changes coming in hot to handle .msi generation for manifest packs.

**Review by jonathanpeppers — APPROVED** [review]
LGTM, but I think @pjcollins wanted to get one more change from a .yaml template before merging this.

**Comment by pjcollins** [comment]
Test run for v2 msi conversion: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5013484&view=results

**Review by mattleibow — APPROVED** [review]

**Pull request merged**
